cdap-ui/app/cdap/components/DataPrep/Directives/MarkAsError/index.js

```diff
@@ -0,0 +1,332 @@
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.offsetCalcDebounce);
+  }
+
+  applyDirective = () => {
+    execute([this.getDirectiveExpression()])
+      .subscribe(() => {
+        this.props.close();
+        this.props.onComplete();
+      }, (err) => {
+        console.log('error', err);
+
+        DataPrepStore.dispatch({
+          type: DataPrepActions.setError,
+          payload: {
+            message: err.message || err.response.message
+          }
+        });
+      });
+  };
+
+  handleTextFilterChange = (e) => {
+    this.setState({textFilter: e.target.value});
+  };
+
+  handleCustomFilterChange = (e) => {
+    this.setState({customFilter: e.target.value});
+  };
+
+  toggleIgnoreCase = () => {
+    this.setState({
+      ignoreCase: !this.state.ignoreCase
+    });
+  };
+
+  getDirectiveExpression = () => {
+    let directive;
+    let condition = 'send-to-error';
+    let column = this.props.column;
+    let textValue = this.state.textFilter;
+    let configuration;
+
+    switch (this.state.selectedCondition) {
+      case 'EMPTY':
+        directive = `${condition} empty(${column})`;
+        break;
+      case 'TEXTCONTAINS':
+        if (this.state.ignoreCase) {
+          textValue = `(?i)${textValue}`;
+        }
+        directive = `${condition} "${column}" =~ "${textValue}.*"`;
+        break;
+      case 'TEXTSTARTSWITH':
+        configuration = `"${textValue}"`;
+        if (this.state.ignoreCase) {
+          column = `${column}.toLowerCase()`;
+          configuration = `"${textValue}".toLowerCase()`;
+        }
+        directive = `${condition} "${column}" =^ "${configuration}"`;
+        break;
+      case 'TEXTENDSWITH':
+        configuration = `"${textValue}"`;
+        if (this.state.ignoreCase) {
+          column = `${column}.toLowerCase()`;
+          configuration = `"${textValue}".toLowerCase()`;
+        }
+        directive = `${condition} "${column}" =$ "${configuration}"`;
+        break;
+      case 'TEXTEXACTLY':
+        if (this.state.ignoreCase) {
+          textValue = `(?i)${textValue}`;
+        }
+        directive = `${condition} "${column}" == "${textValue}$"`;
+        break;
+      case 'TEXTREGEX':
+        directive = `${condition} "${column}" "${textValue}"`;
+        break;
+      case 'CUSTOMCONDITION':
+        directive = `${condition} "${column}" "${this.state.customFilter}"`;
+        break;
+    }
+    return directive;
+  }
+
+  renderTextFilter = () => {
+    if (this.state.selectedCondition.substr(0, 4) !== 'TEXT') { return null; }
+
+    let ignoreCase;
+    if (this.state.selectedCondition !== 'TEXTREGEX') {
+      ignoreCase = (
+        <div>
+          <span
+            className="cursor-pointer"
+            onClick={this.toggleIgnoreCase}
+          >
+            <span
+              className={classnames('fa', {
+                'fa-square-o': !this.state.ignoreCase,
+                'fa-check-square': this.state.ignoreCase
```

Comment:
Can you use `IconSVG` over here? Let's try to use it in future DataPrep directives.